# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/android/AndroidCompositorWidget.cpp
# Commit: b199b21ed5a3
# Full Hash: b199b21ed5a30333c21cac3a2627c9f2ca09b181
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2022-03-09 09:44:44
# Description:
#   Bug 1758217 - Clamp maximum size of AndroidCompositorWidget. r=gfx-reviewers,nical
#   
#   In bug 1756700 we added size constraints to the size of the android
#   nsWindow. This avoided hitting webrender's window_size_sanity_check
#   assertion during the geckoview giantScreenshot test when initializing
# ==============================================================================

diff -r f24e597defe1 -r b199b21ed5a3 widget/android/AndroidCompositorWidget.cpp
--- a/widget/android/AndroidCompositorWidget.cpp	Tue Mar 08 20:21:41 2022 +0000
+++ b/widget/android/AndroidCompositorWidget.cpp	Tue Mar 08 21:00:30 2022 +0000
@@ -93,7 +93,8 @@
 
   const int32_t width = ANativeWindow_getWidth(nativeWindow);
   const int32_t height = ANativeWindow_getHeight(nativeWindow);
-  mClientSize = LayoutDeviceIntSize(width, height);
+  mClientSize = LayoutDeviceIntSize(std::min(width, MOZ_WIDGET_MAX_SIZE),
+                                    std::min(height, MOZ_WIDGET_MAX_SIZE));
 
   ANativeWindow_release(nativeWindow);
 
